### DISCUSSION TITLE
Nextjs API and API call on build.

### DISCUSSION THREAD

**Discussion by Rounak-stha** [vercel/next.js#50454] [App Router]
Data fetching by calling the nextjs API on the top level of the server components is the new pattern now. It all works well during development but what I like to know is which is compiled first during build. We can assume that the `api` must be built first as it can be called by components to fetch data. But I ran into a problem during the build on a server component that fetches data from the same project's API and I am getting an error mentioning `Failed to collect page data`.  
Anyone else facing the same problem ?

**Comment by matheustodao** [comment]
I'm getting similar error

In my case is **AxiosError: connect ECONNREFUSED 127.0.0.1:80**

When i works fine in development but try to build the app i have above issues.

**Reply by Rounak-stha** [reply]
That is because you are trying to connect to localhost (127.0.0.1) on production where you should have used the production URL of your app. 

What you can do is create a field on the env file pointing to the app URL, and use localhost on development and the actual URL on production.

**Reply by matheustodao** [reply]
I already tried to set production url in env file but it didn't work.

If you want to test project, you can see the repository in the branch [example/mock-data](https://github.com/matheustodao/cine-share/blob/example/mock-data) (this is a side-project I'm building)

The file that I instantiated api is [here](https://github.com/matheustodao/cine-share/blob/example/mock-data/src/infra/api/cineShareApi.ts)

**Reply by matheustodao** [reply]
It's the same error when I try to build an app in local development. `yarn build`

But it works fine with `yarn dev`

**Reply by Rounak-stha** [reply]
Somehow the dev url (localhost) is being used during build. Please check the env variables on the settings if you're using vercel.

**Comment by aomini** [comment]
This is probably the case for use of data in the pages. 
Do you get any sort of error or are you able to identify which page is not able to get generated statically?

**Reply by Rounak-stha** [reply]
The blog pages are the ones getting the error. The only error I get is `failed to load page data`. Is there any way to inspect the error in more detail on the build? 
Can you explain how the use case of data might have caused the error? Because it works well in development.

**Comment by Rounak-stha** [comment]
It worked when I deployed the commit with only the API first and then the server component page consuming the API after. Turns out can not build an api and page consuming the api at the same time.
